packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -169,10 +184,30 @@ function s(
     }
     if (
         !isUndefined(slotset) &&
-        !isUndefined(slotset[slotName]) &&
-        slotset[slotName].length !== 0
+        !isUndefined(slotset.slotAssignments) &&
+        !isUndefined(slotset.slotAssignments[slotName]) &&
+        slotset.slotAssignments[slotName].length !== 0
     ) {
-        children = slotset[slotName];
+        children = slotset.slotAssignments[slotName].reduce((acc, vnode) => {
+            // If the passed slot content is factory, evaluate it and use the produced vnodes
+            if (vnode && isVScopedSlotContent(vnode)) {
+                const vmBeingRenderedInception = getVMBeingRendered();
+                let children: VNodes = [];
+                try {
+                    if (!isUndefined(slotset.owner)) {
+                        // Evaluate in the scope of the slot content's owner
+                        setVMBeingRendered(slotset.owner);
+                    }
```

Comment:
Nit: IMO this can be moved outside of the `try`, since if it fails the `VMBeingRendered` will not be set.